src/client/src/App.jsx

```diff
@@ -1,28 +1,117 @@
-import "./App.css";
 import { Routes, Route, BrowserRouter as Router, } from "react-router-dom";
 import React, {useEffect, useState} from "react";
+import { FlashCardPage } from "./FlashCard/Page";
 
-const Placeholder = () => {
-    const [users, setUsers] = useState([]);
 
-    useEffect(() => {
-        fetch('/api/users').then( async (user) => {
-            const usersReturned = await user.json();
-            setUsers(Object.values(usersReturned)[0])
-        })
-    }, []);
+const DUMMY_FLASH_CARDS = [
+  {
+    id: 1,
+    term: 'andrew',
+    definition: 'spencer',
+  },
+];
 
-    return <div>{users.join(',')}</div>
-};
+// Make sure this is unique!
+let nextId = 2;
 
-function App() {
+/**
+ * 
+ * @param {*} props onSubmit function
+ */
+function FlashCardForm(props) {
+  const { onSubmit } = props;
+  // Could use this to edit existing data, so initial values would be here:
+  const [term, setTerm] = useState('');
+  const [definition, setDefinition] = useState('');
+  const [dataIsPosting, setDataIsPosting] = useState(false);
+
+  return (
+    <form onSubmit={(event) => {
+      event.preventDefault();
+      // POST to API, deal with errors, etc. with { term, definition }
+      // Return an object with those values and unique id.
+      // Trigger reload.
+      const id = nextId;
+      nextId += 1;
+      setDataIsPosting(true);
+      // If we add `await onSubmit`, then the button should be disabled.
+      onSubmit({ id, term, definition });
+      setDataIsPosting(false);
+      setTerm('');
+      setDefinition('');
+    }}>
+      <label htmlFor="flashCardTerm">Term:</label>
+      <input id="flashCardTerm" name="term" value={term} onChange={(event) => {
+        setTerm(event.target.value);
+      }} />
+      <label htmlFor="flashCardDefinition">Definition:</label>
+      <input id="flashCardDefinition" name="definition" value={definition} onChange={(event) => {
+        setDefinition(event.target.value);
+      }} />
+      <button type="submit" disabled={dataIsPosting}>Add New Flash Card</button>
+    </form>
+  )
+}
 
+const FlashCards = () => {
```

Comment:
This should be moved to its own file, and refactored to be split up - separate the concerns.
API hook and state management should also be extracted - Move to `react-query`